packages/manager/src/views/WebPropertyEnvPage/components/AddMembers/AddMembers.tsx

```diff
@@ -0,0 +1,620 @@
+      }
+      return temp;
+    });
+  };
+
+  const onSelectAccess = (
+    _event: React.MouseEvent | React.ChangeEvent,
+    selection: string,
+    email: string
+  ) => {
+    setNewUserDetails((prevDetails) => {
+      const temp = [...prevDetails];
+      const index = temp.findIndex((m) => m.email === email);
+      if (index >= 0) {
+        temp[index] = { ...temp[index], isOpen: false, role: selection };
+      }
+      return temp;
+    });
+  };
+
+  const handleAddMemberClick = (
+    _event: React.MouseEvent | React.ChangeEvent,
+    option: RoverItem
+  ) => {
+    setNewUserDetails((prevDetails) => {
+      const temp = [...prevDetails];
+      const check = temp.find((key) => key.name === option.name);
+      if (!check) {
+        temp.push({ ...option, role: 'User', isOpen: false });
+      }
+      return temp;
+    });
+  };
+
+  const onToggleRover = () => {};
+  const onSelectRover: SelectProps['onSelect'] = (_event, selection) => {
+    setUsersData((prevGroup: UserDataTDO) => ({
+      data: prevGroup.data.filter((n: { name: string }) => n.name !== selection)
+    }));
+  };
+
+  const clearSelection = () => {
+    if (activeTabKey === 0) {
+      setSelectedUsers([]);
+      setUsersData({ data: [] });
+      setNewUserDetails([]);
+      setIsOpen(false);
+    } else {
+      setUsersData({ data: [] });
+      setUserListFromRover([]);
+    }
+  };
+
+  const addRoleToRoverGroup = (roverGroupList: RoverItem[]) => {
+    if (roverGroupList) {
+      const updatedRoverGroupList = roverGroupList.map((item: RoverItem) => ({
+        ...item,
+        role: 'User',
+        isOpen: false
+      }));
+      setRoverList(updatedRoverGroupList);
+    } else {
+      toast.error('Please enter a valid Rover Group Common Name (cn)');
+    }
+  };
+
+  const debounceCustomCombined = (
+    delay: number,
+    selectedTab: string | number
+  ): SelectProps['onFilter'] => {
+    let timer: NodeJS.Timeout;
+    return (
+      _e: React.ChangeEvent<HTMLInputElement> | null,
+      value: string
+    ): React.ReactElement[] | undefined => {
+      clearInterval(timer);
+      timer = setTimeout(async () => {
+        if (value.length >= 3) {
+          if (selectedTab === 0) {
+            const res = await fetchUserlist(value);
+            setUserListFromRover(res);
+            setIsOpen(true);
+          } else if (selectedTab === 1) {
+            const res = await fetchRoverGroup(value);
+            addRoleToRoverGroup(res);
+          }
+        }
+      }, delay);
+      return
+    };
+  };
+
+  const useAddPermission1 = useAddPermission(propertyIdentifier);
+  const handleSubmit = () => {
+    const addData: AddDataType = {
+      propertyIdentifier,
+      permissionDetails: usersData.data.map(({ email, name, ...rand }) => ({
+        email,
+        name,
+        actions: Object.keys(rand)
+      }))
+    };
+    useAddPermission1.mutateAsync(addData).then(() => {
```

Comment:
Refer toast error message in edit section

Reply:
Done